src/Adapter/NewProducts/NewProductsProductSearchProvider.php

```diff
@@ -62,21 +62,23 @@ public function __construct(
      * @param ProductSearchQuery $query
      * @param string $type
      *
-     * @return array
+     * @return array|int
      */
     private function getProductsOrCount(
         ProductSearchContext $context,
         ProductSearchQuery $query,
         $type = 'products'
     ) {
-        return Product::getNewProducts(
+        $result = Product::getNewProducts(
```

Comment:
Should not be changed. It's a BC

Reply:
It's not a BC : it's a private function.